src/frontend/resources/general/js/components-pages/CudosBridgePageComponent.tsx

```diff
@@ -286,7 +313,10 @@ export default class CudosBridgeComponent extends ContextPageComponent<Props, St
             balance = new BigNumber(0);
         }
 
-        let maximumAmount = BigNumber.maximum(balance, this.state.contractBalance).minus(this.state.minBridgeFeeAmount);
+        let maximumAmount = BigNumber.maximum(balance, this.state.walletBalance).minus(this.state.minBridgeFeeAmount);
+
+        await this.setSimulatedMsgsCost(maximumAmount.toString());
+        maximumAmount = maximumAmount.minus(this.state.estimatedGasFees);

```

Comment:
I suggest returning estimateGasFees from setSimulatedMsgsCost and you use the this.state object only in UI. Because setState is async operation - https://linguinecode.com/post/why-react-setstate-usestate-does-not-update-immediately